src/renderer/index.tsx

```diff
@@ -1,91 +1,59 @@
-import React, { useState, useEffect, useReducer } from 'react';
-import { connect } from 'react-redux';
+import React, { useState, useEffect } from 'react';
 import { Overview } from './overview';
 import { FileListView } from './fileListView/fileListView';
 import { IPC_EVENTS } from '../constants';
 import { ipcRenderer } from 'electron';
-import { SiteImageData, OptimizerStatus } from '../types';
-import { Preferences } from '../types';
-import { scanImageReducer, initialState, SCAN_IMAGES_ACTIONS } from './reducers/scanImageReducer';
-import * as LocalRenderer from '@getflywheel/local/renderer';
-import { fileListReducer } from './reducers/fileListReducer';
-import { POPULATE_FILE_LIST } from './reducers/fileListReducer';
-import { DatasetType } from '../types';
+import { OptimizerStatus, ImageData } from '../types';
+import { store, actions, selectors, useStoreSelector } from './store';
 
 interface IProps {
-	preferences: Preferences;
 	match: { params: { siteID: string; } };
 }
 
+
 const ImageOptimizer = (props: IProps) => {
+	const { match } = props;
+	const { siteID } = match.params;
+
+	const preferences = useStoreSelector((state) => state.preferences);
+	const siteImageData = useStoreSelector((state) => state.sites[siteID]);
 	const [overviewSelected, setOverviewSelected] = useState(true);
-	const initialImageData = {} as SiteImageData;
-	const [siteImageData, dispatchSiteImageData] = useReducer(fileListReducer, initialImageData);
-	const [scanImageState, dispatchScanImageData] = useReducer(scanImageReducer, initialState);
 
 	const scanForImages = () => {
-		dispatchScanImageData({ type: SCAN_IMAGES_ACTIONS.REQUEST });
+		store.dispatch(actions.scanRequest(siteID));
+
 		ipcRenderer.send(
 			IPC_EVENTS.SCAN_FOR_IMAGES,
-			props.match.params.siteID
+			siteID,
 		);
 	}
 
-	// set up initial state for file list view
-	useEffect(
-		() => {
-			const initialImageScan = async () => {
-				await fetchImageStateData();
-			}
-			initialImageScan();
-		}, []
-	);
-
-	const fetchImageStateData = async () => {
-		const mainImageData = await LocalRenderer.ipcAsync(
-			IPC_EVENTS.GET_IMAGE_DATA,
-			props.match.params.siteID,
-			DatasetType.ONLY_UNCOMPRESSED
-		);
-
-		dispatchScanImageData({ type: SCAN_IMAGES_ACTIONS.SUCCESS, payload: mainImageData });
-
-		dispatchSiteImageData({
-			type: POPULATE_FILE_LIST.SET_IMAGE_DATA, payload: mainImageData
-		});
-	}
 	// listen for optimization events and update status accordingly
 	useEffect(
 		() => {
 			if (!ipcRenderer.listenerCount(IPC_EVENTS.COMPRESS_IMAGE_SUCCESS)) {
 				ipcRenderer.on(
 					IPC_EVENTS.COMPRESS_IMAGE_SUCCESS,
-					(_, newImageData: ImageData) => {
-						dispatchSiteImageData({
-							type: POPULATE_FILE_LIST.IMAGE_OPTIMIZE_SUCCESS, payload: newImageData
-						});
+					(_, imageData: ImageData) => {
+						store.dispatch(actions.optimizeSuccess({ siteID, imageData }));
```

Comment:
Did you end up making the decision to leave this IPC listeners the way they are?

Reply:
Yes, but only because it seemed out of scope for this guy. I've got a branch started that pulls these out for LOC-2232